types/sanctuary/index.d.ts

```diff
@@ -96,16 +96,17 @@ declare namespace Sanctuary {
       version: NonNegativeInteger
     };
     is(typeRep: TypeRep): (x: any) => boolean;
-    //  Showable
-    toString(x: any): string;
+    // Showable
+    show(x: any): string;
     //  Fantasy Land
     equals<A>(x: Setoid<A>): (y: Setoid<A>) => boolean;
     lt <A>(x: Ord<A>): (y: Ord<A>) => boolean;
     lte<A>(x: Ord<A>): (y: Ord<A>) => boolean;
     gt <A>(x: Ord<A>): (y: Ord<A>) => boolean;
     gte<A>(x: Ord<A>): (y: Ord<A>) => boolean;
-    min<A>(x: Ord<A>): (y: Ord<A>) => A;
-    max<A>(x: Ord<A>): (y: Ord<A>) => A;
+    min<A>(x: A): (y: A) => A;
+    max<A>(x: A): (y: A) => A;
```

Comment:
Why `A` rather than `Ord<A>`?

Reply:
Using the previous declaration the expression `S.min(0)(100)` will have type `unknown`, as the type system from TypeScript is not able to infer the type = /
I also tried to declare the variable in all functions, but it didn't work either (as in `min<A>(x: Ord<A>): <A>(y: Ord<A>) => A;`)
That's a point where we may choose to be "ts correct" or "type correct" - having some `unknown`s in the code can trigger errors in some expressions in the ts compiler - but in the other hand, using that generic will not stop the user from providing something that is not an `Ord` to the function (but sanctuary's type-checker will catch those). 
Looking at what other libs are doing, I saw that the way the "solve" this [is to use just `A`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ramda/index.d.ts#L1044) to circumvent this limitation.

Reply:
Managed to make the type inference work by replacing
```
interface Setoid<A> {}
interface Ord<A> extends Setoid<A> {}
```
with
```
type Setoid<A> = Ord<A>
type Ord<A> = A
```
But I think that this would also require rewriting most of the basic declarations 🤔

Reply:
Thanks for the explanation, @lfarroco. Let's stick with `A` for now, for the reason you stated. If you feel ambitious you could try your new approach in a stand-alone pull request. :)